### PR TITLE
Richard rodriguez/single player mode

### PR DESCRIPTION
**Pull request by rmjuarez12**
# Pull Request Template

Go from a screen that selects the game play of single player or multiplayer. 
When the single player mode is selected it goes into the screen which gives the option to read, write, and/or draw. 

    1. You will be able to read, draw, and write. 
    2. You can choose either draw/write, or both
    3. One must at least be uploaded

When it is submitted, a function will put the player against a randomly chosen boss(bot). 

# Things that changed

I have added a new folder on the Components > Pages directory under the name "Gameification". This directory holds the main process of the game, which is the Read, Draw, Write. For the uploading of the files, I am working with the component "Upload Docs", which is under Components > Common. I made a few minor adjustments to this component:
- Added "listType", "handleChangeExtra", and "savedFileList" props
- "handleChangeExtra" is being used on the handleChange function. This is so that I can also have access to that event
- "savedFileList" is being used in the useffect, and this is to auto-populate the preview file list state

## Commits Checklist
- [x ] commits have clear title names 
- [x ] commit has descriptive message of what has changed 
- [ ] more small commits than less big commits 

## Code Checklist 
- [ x] code is formatted appropriately 
- [x ] code meets DRY standards
- [x ] no dead code (check for logs and print statements)  
- [x ] no TODOS instead necessary comments 
- [x ] code follows Story Squad standards for:
     - Language 
     - Framework 
     - Libraries   

### PRs needs to be reviewed by at least 2 team members
 - TPL is the 3rd reviewer 
 - Lots of comments and suggestions, please! 

[Pull Request Rubric](https://www.notion.so/1fc04e4fedeb429ba873b7c68d281707?v=74054da7991341c0bf970f39410c43da)

### CODE REVIEW

**Review by derekjpeters — DISMISSED** [review]
Richard, Code actually looks really clean.  I was able to visually see what you were adding.  I approve these changes

**Comment by derekjpeters** [comment]
Looks like you need 1 more review.

File: src/components/pages/Gameification/GameModal.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    gsap.to('#notification-modal', { opacity: 0, duration: 0.5, delay: 0.5 });
+    gsap.to('#notification-modal', {
+      display: 'none',
+      duration: 0.1,
+      delay: 1,
+    });
+
+    setTimeout(() => {
+      props.disableModalWindow();
+    }, 1000);
```
Comment (anilgitme) [inline src/components/pages/Gameification/GameModal.js]: Try to see if the value 1000 can be moved to a constant variable.

File: src/components/pages/Gameification/GameWriteStep.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        title: 'Get ready for the boss battle!',
+        description:
+          'To complete this mission, your work will be put up head-to-head against a boss. If you are ready to fight, submit your work!',
+        buttonTxt: "Let's Go!",
+      };
+
+      props.enableModalWindow(modalData);
+
+      setIsUploading(false);
+    }, 1500);
```
Comment (anilgitme) [inline src/components/pages/Gameification/GameWriteStep.js]: Try to use constant variables for timeout values

**Review by anilgitme — DISMISSED** [review]
Changes looks good. Hope we added some tests.

File: src/components/pages/Gameification/GameModal.js (hunk 1 above)
Comment (rmjuarez12) [reply src/components/pages/Gameification/GameModal.js]: Ok, got it

File: src/components/pages/Gameification/GameWriteStep.js (hunk 2 above)
Comment (rmjuarez12) [reply src/components/pages/Gameification/GameWriteStep.js]: Got it

File: src/components/pages/Gameification/GameModal.js (hunk 1 above)
Comment (rmjuarez12) [reply src/components/pages/Gameification/GameModal.js]: Alrighty. done

File: src/components/pages/Gameification/GameWriteStep.js (hunk 2 above)
Comment (rmjuarez12) [reply src/components/pages/Gameification/GameWriteStep.js]: Done.

**Review by derekjpeters — APPROVED** [review]

**Review by anilgitme — APPROVED** [review]
Looks good!

**Pull request merged**